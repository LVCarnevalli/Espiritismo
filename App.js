import { ActionSheetProvider } from '@expo/react-native-action-sheet';
import * as Icon from '@expo/vector-icons';
import { AppLoading } from 'expo';
import { Asset } from 'expo-asset';
import * as Font from 'expo-font';
import * as firebase from 'firebase';
import React from 'react';
import { Platform, StatusBar, StyleSheet, View } from 'react-native';
import {
  FIREBASE_API_KEY,
  FIREBASE_AUTH_DOMAIN,
  FIREBASE_DATABASE_URL,
  FIREBASE_PROJECT_ID,
} from 'react-native-dotenv';
import ignoreWarnings from 'react-native-ignore-warnings';
import { Provider } from 'react-redux';
import { PersistGate } from 'redux-persist/integration/react';

import AppNavigator from './src/navigation/SwitchNavigator';
import configureStore from './src/store/Store';

const { persistor, store } = configureStore();

ignoreWarnings('Setting a timer');
ignoreWarnings('componentWillReceiveProps has been renamed');
ignoreWarnings('componentWillMount has been renamed');
ignoreWarnings('componentWillUpdate has been renamed');

class App extends React.Component {
  static defaultProps = {
    skipLoadingScreen: false,
  };

  state = {
    isLoadingComplete: false,
  };

  constructor(props) {
    super(props);

    if (!firebase.apps.length) {
      firebase.initializeApp({
        apiKey: FIREBASE_API_KEY,
        authDomain: FIREBASE_AUTH_DOMAIN,
        databaseURL: FIREBASE_DATABASE_URL,
        projectId: FIREBASE_PROJECT_ID,
      });
    }
  }

  _loadResourcesAsync = async () => {
    return Promise.all([
      Asset.loadAsync([
        require('./assets/images/BgSun.png'),
        require('./assets/images/BgBlood.png'),
        require('./assets/images/BgMoon.png'),
        require('./assets/images/IconGesture.png'),
        require('./assets/images/IconLightbulb.png'),
        require('./assets/images/IconQuestion.png'),
        require('./assets/images/IconSearch.png'),
        require('./assets/images/IconRead.png'),
        require('./assets/images/IconSave.png'),
        require('./assets/images/IconLove.png'),
      ]),
      Font.loadAsync({
        ...Icon.Feather.font,
        'open-sans-bold': require('./assets/fonts/OpenSans/OpenSans-Bold.ttf'),
        'open-sans-extra-bold': require('./assets/fonts/OpenSans/OpenSans-ExtraBold.ttf'),
        'open-sans-regular': require('./assets/fonts/OpenSans/OpenSans-Regular.ttf'),
        'open-sans-italic': require('./assets/fonts/OpenSans/OpenSans-Italic.ttf'),
        'open-sans-light': require('./assets/fonts/OpenSans/OpenSans-Light.ttf'),
        'open-sans-semi-bold': require('./assets/fonts/OpenSans/OpenSans-SemiBold.ttf'),
        'grotes-sans-regular': require('./assets/fonts/GrotesSansDemoRegular.ttf'),
      }),
    ]);
  };

  _handleLoadingError = error => {
    // In this case, you might want to report the error to your error
    // reporting service, for example Sentry
    console.warn(error);
  };

  _handleFinishLoading = () => {
    this.setState({ isLoadingComplete: true });
  };

  render() {
    if (!this.state.isLoadingComplete && !this.props.skipLoadingScreen) {
      return (
        <AppLoading
          startAsync={this._loadResourcesAsync}
          onError={this._handleLoadingError}
          onFinish={this._handleFinishLoading}
        />
      );
    } else {
      return (
        <View style={styles.container}>
          {Platform.OS === 'ios' && <StatusBar barStyle="default" />}
          <Provider store={store}>
            <PersistGate loading={null} persistor={persistor}>
              <ActionSheetProvider>
                <AppNavigator />
              </ActionSheetProvider>
            </PersistGate>
          </Provider>
        </View>
      );
    }
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
  },
});

export default App;
