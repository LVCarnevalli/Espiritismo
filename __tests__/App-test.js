import React from 'react';
import NavigationTestUtils from 'react-navigation/NavigationTestUtils';
import renderer from 'react-test-renderer';

import App from '../App';

describe('App snapshot', () => {
  jest.useFakeTimers();
  beforeEach(() => {
    NavigationTestUtils.resetInternalState();
  });

  test('renders the loading screen', async () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders the root without loading screen', async () => {
    const tree = renderer.create(<App skipLoadingScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
