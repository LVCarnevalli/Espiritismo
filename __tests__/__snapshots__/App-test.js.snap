// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot renders the loading screen 1`] = `null`;

exports[`App snapshot renders the root without loading screen 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        onMoveShouldSetResponder={[Function]}
        onMoveShouldSetResponderCapture={[Function]}
        onResponderEnd={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderReject={[Function]}
        onResponderRelease={[Function]}
        onResponderStart={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        onStartShouldSetResponderCapture={[Function]}
        style={
          Array [
            Object {
              "flex": 1,
              "flexDirection": "column-reverse",
              "overflow": "hidden",
            },
            Object {
              "backgroundColor": "#000",
            },
          ]
        }
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            accessibilityElementsHidden={false}
            pointerEvents="auto"
            style={
              Object {
                "backgroundColor": "#E9E9EF",
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "paddingTop": 0,
                "position": "absolute",
                "right": 0,
                "shadowColor": "black",
                "shadowOffset": Object {
                  "height": 0,
                  "width": 0,
                },
                "shadowOpacity": 0.2,
                "shadowRadius": 5,
                "top": 0,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                  Object {
                    "translateY": 0,
                  },
                ],
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <View
                accessibilityIgnoresInvertColors={true}
                style={
                  Object {
                    "height": "100%",
                    "width": "100%",
                  }
                }
              >
                <Image
                  source={1}
                  style={
                    Array [
                      Object {
                        "bottom": 0,
                        "left": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      },
                      Object {
                        "height": "100%",
                        "width": "100%",
                      },
                      undefined,
                    ]
                  }
                />
                <View
                  style={
                    Object {
                      "height": "20%",
                    }
                  }
                />
                <View
                  style={
                    Object {
                      "flex": 1,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "white",
                          "fontFamily": "grotes-sans-regular",
                          "fontSize": 50,
                          "paddingLeft": 30,
                          "paddingRight": 30,
                          "textAlign": "center",
                        }
                      }
                    >
                      espiritismo
                    </Text>
                  </View>
                </View>
                <View
                  style={
                    Object {
                      "height": 70,
                      "width": "100%",
                    }
                  }
                >
                  <View
                    accessible={true}
                    focusable={true}
                    isTVSelectable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "alignItems": "center",
                        "borderBottomWidth": 0,
                        "borderColor": "white",
                        "borderLeftWidth": 0,
                        "borderRightWidth": 0,
                        "borderStyle": "solid",
                        "borderTopWidth": 1,
                        "height": "100%",
                        "justifyContent": "center",
                        "opacity": 0.2,
                        "width": "100%",
                      }
                    }
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "white",
                            "fontSize": 20,
                            "fontWeight": "bold",
                          },
                          Object {
                            "fontFamily": "open-sans-bold",
                          },
                        ]
                      }
                    >
                      CARREGANDO
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
